packages/braintree-integration/src/braintree-visa-checkout/braintree-visa-checkout-customer-strategy.ts

```diff
@@ -0,0 +1,253 @@
+import { FormPoster } from '@bigcommerce/form-poster';
+
+import {
+    BraintreeDataCollector,
+    BraintreeSdk,
+    BraintreeVisaCheckout,
+    VisaCheckoutAddress,
+    VisaCheckoutPaymentSuccessPayload,
+    VisaCheckoutTokenizedPayload,
+} from '@bigcommerce/checkout-sdk/braintree-utils';
+import {
+    Address,
+    CustomerInitializeOptions,
+    CustomerStrategy,
+    ExecutePaymentMethodCheckoutOptions,
+    InvalidArgumentError,
+    LegacyAddress,
+    MissingDataError,
+    MissingDataErrorType,
+    NotImplementedError,
+    PaymentIntegrationService,
+    RequestOptions,
+} from '@bigcommerce/checkout-sdk/payment-integration-api';
+
+import { WithBraintreeVisaCheckoutCustomerInitializeOptions } from './braintree-visa-customer-initialize-options';
+
+export default class BraintreeVisaCheckoutCustomerStrategy implements CustomerStrategy {
+    private buttonClassName = 'visa-checkout-wrapper';
+
+    constructor(
+        private paymentIntegrationService: PaymentIntegrationService,
+        private formPoster: FormPoster,
+        private braintreeSdk: BraintreeSdk,
+    ) {}
+
+    async initialize(
+        options: CustomerInitializeOptions & WithBraintreeVisaCheckoutCustomerInitializeOptions,
+    ): Promise<void> {
+        const { braintreevisacheckout: visaCheckoutOptions, methodId } = options;
+
+        if (!visaCheckoutOptions || !methodId) {
+            throw new InvalidArgumentError(
+                'Unable to proceed because "options.braintreevisacheckout" argument is not provided.',
+            );
+        }
```

Comment:
Have 2 points around this condition:
1. It might be better to separate this condition an throw different errors. There might be a case, when visaCheckoutOptions provides an empty object (so it is exists as a value), however methodId is undefined, therefore "Unable to proceed because "options.braintreevisacheckout" argument is not provided" is not quite right in such situation.
2. Also, we should update visaCheckoutOptions check to make sure that `container` option was provided, since if it's not (empty object was passed), then it might brake the implementation